sklearn/covariance/_graph_lasso.py

```diff
@@ -88,112 +69,17 @@ def graphical_lasso(
-        direction. This parameter controls the accuracy of the search direction
-        for a given column update, not of the overall parameter estimate. Only
-        used for mode='cd'. Range is (0, inf].
-
-    max_iter : int, default=100
-        The maximum number of iterations.
-
-    verbose : bool, default=False
-        If verbose is True, the objective function and dual gap are
-        printed at each iteration.
-
-    return_costs : bool, default=False
-        If return_costs is True, the objective function and dual gap
-        at each iteration are returned.
-
-    eps : float, default=eps
-        The machine-precision regularization in the computation of the
-        Cholesky diagonal factors. Increase this for very ill-conditioned
-        systems. Default is `np.finfo(np.float64).eps`.
-
-    return_n_iter : bool, default=False
-        Whether or not to return the number of iterations.
-
-    Returns
-    -------
-    covariance : ndarray of shape (n_features, n_features)
-        The estimated covariance matrix.
-
-    precision : ndarray of shape (n_features, n_features)
-        The estimated (sparse) precision matrix.
-
-    costs : list of (objective, dual_gap) pairs
-        The list of values of the objective function and the dual gap at
-        each iteration. Returned only if return_costs is True.
-
-    n_iter : int
-        Number of iterations. Returned only if `return_n_iter` is set to True.
-
-    See Also
-    --------
-    GraphicalLasso : Sparse inverse covariance estimation
-        with an l1-penalized estimator.
-    GraphicalLassoCV : Sparse inverse covariance with
-        cross-validated choice of the l1 penalty.
-
-    Notes
-    -----
-    The algorithm employed to solve this problem is the GLasso algorithm,
-    from the Friedman 2008 Biostatistics paper. It is the same algorithm
-    as in the R `glasso` package.
-
-    One possible difference with the `glasso` R package is that the
-    diagonal coefficients are not penalized.
-    """
     _, n_features = emp_cov.shape
     if alpha == 0:
-        if return_costs:
-            precision_ = linalg.inv(emp_cov)
-            cost = -2.0 * log_likelihood(emp_cov, precision_)
-            cost += n_features * np.log(2 * np.pi)
-            d_gap = np.sum(emp_cov * precision_) - n_features
-            if return_n_iter:
-                return emp_cov, precision_, (cost, d_gap), 0
-            else:
-                return emp_cov, precision_, (cost, d_gap)
-        else:
-            if return_n_iter:
-                return emp_cov, linalg.inv(emp_cov), 0
-            else:
-                return emp_cov, linalg.inv(emp_cov)
+        # Early return without regularization
+        precision_ = linalg.inv(emp_cov)
+        cost = -2.0 * log_likelihood(emp_cov, precision_)
+        cost += n_features * np.log(2 * np.pi)
+        d_gap = np.sum(emp_cov * precision_) - n_features
+        return emp_cov, precision_, (cost, d_gap), 0
+
```

Comment:
In line 99, could you add:

```python
i = 0  # initialize the counter to be robust to `max_iter=0`
```

It will solve the issue. This case happen with `GraphicalLassoCV`.

Reply:
Okay, this has now been updated